src/journal/journald-file.c

```diff
@@ -106,9 +119,12 @@ static int journald_file_punch_holes(JournalFile *f) {
         sz = le64toh(f->header->data_hash_table_size);
 
         for (uint64_t i = p; i < p + sz && n > 0; i += n) {
-                n = pread(f->fd, items, MIN(sizeof(items), p + sz - i), i);
+                size_t m = MIN(sizeof(items), p + sz - i);
+                n = pread(f->fd, items, m, i);
                 if (n < 0)
-                        return n;
+                        return log_debug_errno(errno, "Failed to read hash table items: %m");
+                if ((size_t) n != m)
+                        return log_debug_errno(SYNTHETIC_ERRNO(EIO), "Short read while reading hash table items.");
```

Comment:
I'm not sure we should fail on short reads here. I think we already handle those correctly since we check for partially read hash items and exit if we end up not reading anything at all.

Reply:
hmm, yeah, you actually make use of `n` there, true. will rework.